external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSourceProvider.scala

```diff
@@ -564,20 +575,38 @@ private[kafka010] object KafkaSourceProvider extends Logging {
 
   def getKafkaOffsetRangeLimit(
       params: CaseInsensitiveMap[String],
+      globalOffsetTimestampOptionKey: String,
       offsetByTimestampOptionKey: String,
       offsetOptionKey: String,
       defaultOffsets: KafkaOffsetRangeLimit): KafkaOffsetRangeLimit = {
-    params.get(offsetByTimestampOptionKey).map(_.trim) match {
-      case Some(json) => SpecificTimestampRangeLimit(JsonUtils.partitionTimestamps(json))
-      case None =>
-        params.get(offsetOptionKey).map(_.trim) match {
-          case Some(offset) if offset.toLowerCase(Locale.ROOT) == "latest" =>
-            LatestOffsetRangeLimit
-          case Some(offset) if offset.toLowerCase(Locale.ROOT) == "earliest" =>
-            EarliestOffsetRangeLimit
-          case Some(json) => SpecificOffsetRangeLimit(JsonUtils.partitionOffsets(json))
-          case None => defaultOffsets
-        }
+    // The order below represents "preferences"
+
+    // 1. global timestamp
+    if (params.contains(globalOffsetTimestampOptionKey)) {
+      val tsStr = params(globalOffsetTimestampOptionKey).trim
+      try {
+        val ts = tsStr.toLong
+        return GlobalTimestampRangeLimit(ts)
```

Comment:
Super nit: If we put case 2 and 3 into else then we don't need return statement.

Reply:
I intended to avoid deeper level of indentations, but if ~ else if ~ else would achieve the same without return. Will address.